code/modules/assembly/proximity.dm

```diff
@@ -36,29 +41,28 @@
 	return TRUE
 
 /obj/item/assembly/prox_sensor/on_detach()
-	. = ..()
-	if(!.)
-		return
-	else
-		proximity_monitor.SetHost(src,src)
+	proximity_monitor.SetHost(src, src)
+
+/obj/item/assembly/prox_sensor/on_attach()
+	proximity_monitor.SetHost(loc, src)
 
 /obj/item/assembly/prox_sensor/toggle_secure()
 	secured = !secured
 	if(!secured)
-		if(scanning)
-			toggle_scan()
-			proximity_monitor.SetHost(src,src)
+		toggle_scan(FALSE)
 		timing = FALSE
 		STOP_PROCESSING(SSobj, src)
 	else
 		START_PROCESSING(SSobj, src)
-		proximity_monitor.SetHost(loc,src)
 	update_appearance()
 	return secured
 
 /obj/item/assembly/prox_sensor/HasProximity(atom/movable/AM as mob|obj)
-	if (istype(AM, /obj/effect/beam))
+	if(AM == src || AM == loc) // no self or holder
 		return
+	if(is_type_in_typecache(AM, ignore_types))
+		return
```

Comment:
Turn this into a redefined proc, trait or variable, instead of a type cache, please.
Blacklists are not very maintainable.